h2o-algos/src/main/java/hex/tree/Score.java

```diff
@@ -88,8 +98,12 @@ else if (_sii != null)
       if( nclass > 1 ) cdists[0] = GenModel.getPrediction(cdists, m._output._priorClassDist, tmp, m.defaultThreshold()); // Fill in prediction
       val[0] = (float)ys.atd(row);
       _mb.perRow(cdists, val, weight, offset, m);
+
       if (_preds != null)
         _mb.cachePrediction(cdists, allchks, row, chks.length, m);
+
+      // Compute custom metric if necessary
+      customMetricPerRow(cdists, val, weight, offset, m);
```

Comment:
This seems to have the same API as perRow() function. This custom logic seems out-of-place here in the Score task. It seems like it would better fit in the MetricBuilder itself.

What is the intention here?